integrations/uphold/src/test/java/org/dash/wallet/integration/uphold/UpholdClientTest.kt

```diff
@@ -0,0 +1,142 @@
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.mockito.MockitoAnnotations
+import java.net.HttpURLConnection
+
+class UpholdClientTest {
+    private lateinit var mockWebServer: MockWebServer
+    private lateinit var api: UpholdService
+
+    private val capabilitiesJson = """
+        {
+            "category": "permissions",
+            "enabled": true,
+            "key": "withdrawals",
+            "name": "Withdrawals",
+            "requirements": ["user-must-submit-identity"],
+            "restrictions": []
+        }
+    """
+
+    private val cardJson = """
+        [{
+          "address": {
+            "dash": "XjBya4EnibUyxubEA8D2Y8KSrBMW1oHq5U",
+            "wire": "some id"
+          },
+          "available": "45.00",
+          "balance": "45.00",
+          "currency": "DASH",
+          "id": "024e51fc-5513-4d82-882c-9b22024280cc",
+          "label": "DASH card",
+          "lastTransactionAt": "2018-08-01T09:53:44.617Z",
+          "normalized": [{
+            "available": "4500.00",
+            "balance": "4500.00",
+            "currency": "USD"
+          }],
+          "settings": {
+            "position": 1,
+            "protected": false,
+            "starred": true
+          }
+        },
+        {
+          "address": {
+            "bitcoin": "ms22VBPSahNTxHZNkYo2d4Rmw1Tgfx6ojr"
+          },
+          "available": "146.38",
+          "balance": "146.38",
+          "currency": "EUR",
+          "id": "bc9b3911-4bc1-4c6d-ac05-0ae87dcfc9b3",
+          "label": "EUR card",
+          "lastTransactionAt": "2018-08-01T09:53:51.258Z",
+          "normalized": [{
+            "available": "170.96",
+            "balance": "170.96",
+            "currency": "USD"
+          }],
+          "settings": {
+            "position": 2,
+            "protected": false,
+            "starred": true
+          }
+        }]
+    """
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        mockWebServer = MockWebServer()
+        mockWebServer.start(8080)
+        api = RemoteDataSource().buildApi(UpholdService::class.java, "http://127.0.0.1:8080", OkHttpClient())
+    }
+
+    @After
+    fun teardown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun getCapabilities_returnsCapabilitiesObject() {
+        val response = MockResponse()
+            .setResponseCode(HttpURLConnection.HTTP_OK)
+            .setBody(capabilitiesJson)
+        mockWebServer.enqueue(response)
+
+        val expected = UpholdCapability(
+            "withdrawals",
+            "Withdrawals",
+            "permissions",
+            true,
+            listOf("user-must-submit-identity"),
+            listOf()
+        )
+
+        val actualResponse = api.getCapabilities("withdrawals").execute()
+        assertEquals(expected, actualResponse.body())
+    }
+
+    @Test
+    fun getCards_returnsCardList() {
```

Comment:
These tests are not terribly useful since they basically test the moshi parser in Retrofit - I needed it to test my changes with `UpholdAddressAdapter`. In the future, we can build more tests on top of these, although it would require some more refactoring of `UpholdClient`.